termlink/snomedct.py

```diff
@@ -0,0 +1,143 @@
+"""Handles Snomed-CT conversion.
+
+This module provides methods to extract, transform and load relationships
+defined by the snomed-ct dataset.
+
+The download files for snomed-ct are provided at https://www.nlm.nih.gov/healthit/snomedct/.
+"""
+import petl as etl
+import json
+import os
+
+from termlink.commands import SubCommand
+from termlink.models import Coding, Relationship, RelationshipSchema
+from termlink.services import RelationshipService
+from os import listdir
+from dataclasses import replace
+from urllib.parse import urlparse
+
+_STATED_RELATIONSHIP_FILE_FIELDS = ["id", "effectiveTime", "active", "moduleId", "sourceId", "destinationId",
+                                    "relationshipGroup", "typeId", "characteristicTypeId", "modifierId"]
+_DESCRIPTION_FILE_FIELDS = ["id", "effectiveTime", "active", "moduleId", "conceptId", "languageCode", "typeId",
+                            "term", "caseSignificanceId"]
+_IS_A_TYPE_ID = "116680003"
+_FULLY_SPECIFIED_NAME_TYPE_ID = "900000000000003001"
+_FIRST_LETTER_CAPITAL_CASE_ID = "900000000000020002"
+
+
+def _to_json(rec):
+    """
+    Convert record in table to Relationship as a JSON object
+
+    Record is expected to have the following fields: [ source.CODE, source.STR, target.CODE, target.STR]
+
+    Args:
+        rec: A table record
+
+    Returns:
+        A new record containing a single field, which is the JSON object
+    """
+
+    source = Coding(
+        system="http://snomed.info/sct",
+        code=rec['sourceId'],
+        display=rec['source.term']
+    )
+
+    target = Coding(
+        system="http://snomed.info/sct",
+        code=rec['destinationId'],
+        display=rec['target.term']
+    )
+
+    if rec['include-version-mapping']:
```

Comment:
Is replace a copy operation? If so, it would be faster to use `version = rec['effectiveTime'] if rec['include-version-mapping'] else None` and then insert `version` into the `Coding` construction

Reply:
+1